meta-store/src/object_storage/s3_compatible.rs

```diff
@@ -0,0 +1,80 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use std::ops::Range;
+
+use async_trait::async_trait;
+use rusoto_core::Region;
+
+use quickwit_storage::S3CompatibleObjectStorage;
+
+use crate::meta_store::{IndexId, MetaStore, MetaStoreResult, SplitId, SplitManifest, State};
+
+/// S3 Compatible object storage implementation.
+#[derive(Debug)]
+pub struct S3CompatibleMetaStore {
```

Comment:
This is too specific.

We do not want a S3CompatibleMetaStore... Instead we want a `SingleFileMetaStore` that wraps a `Box<dyn Storage>`.

The benefit is very important:
- other projects will be able to use the ram storage for unit tests.
- we will eventually be able to handle other object storages (non S3 compatible) in the future
- we will be able to use the local filesystem
- we can rely on the storage uri resolver in the CLI.

Reply:
@fulmicoton 
Changed it to a single file meta store.